### DISCUSSION TITLE
Conditionally trigger a job based on a resource NOT being updated by another job

### DISCUSSION THREAD

**Discussion by dennisjbell** [concourse/concourse#7278] [Help & Support] [answered]
This has to be a common, possibly solved, problem.  I have a pipeline that detects changes in a git repo, builds a test build, runs tests, then releases that build if it passes.  The release also updates the git repo, which triggers the process to start again.

I want to have the first step basically say trigger on any change to the git repo that DIDN'T come from the release job.  The way I would envision this is a not-passed condition:

```
- get: git
  trigger: true
  passed:
    - ! release
```

Is this already possible through another mechanism?  I couldn't find any docs that would indicate it is.

The other way I though of doing it is a lot more convoluted.  Basically have a hidden "filter" job in another group that triggers on any git repo change, but only passes if it doesn't have the commit message pattern associated with the release, and then make the "first" job trigger on the git resource that passed the filter job.  The problem with this is that its an extra step that shows failed every time there's a release, so it feels dirty.  But this would work today with no changes needed to concourse.

Is there another way that people accomplish this filtering to prevent cycles?

**Comment by taylorsilva** [comment, marked as answer]
You could configure the git resource (assuming you're using that) to ignore commits based on one of the following:

- the path, using `ignore_paths`
- commits labelled with `[ci skip]` using `disable_ci_skip`

See the ReadMe of the resource for details on those params: https://github.com/concourse/git-resource

I have run into this pain too though. The problem is with the schedulers algorithm, it's not smart enough to avoid this loop that a job can put itself into when `get`ing and `put`ing to the same resource. There's probably an issue open for this somewhere.
